Userland/Libraries/LibGL/Tex/TextureUnit.cpp

```diff
@@ -14,20 +14,23 @@ void TextureUnit::bind_texture_to_target(GLenum texture_target, const RefPtr<Tex
     switch (texture_target) {
     case GL_TEXTURE_2D:
         m_texture_target_2d = static_ptr_cast<Texture2D>(texture);
-        m_currently_bound_texture = texture;
         m_currently_bound_target = GL_TEXTURE_2D;
+        m_texture_2d_enabled = true;
         break;
     default:
         VERIFY_NOT_REACHED();
     }
+
+    m_currently_bound_texture = texture;
 }
 
 void TextureUnit::unbind_texture(GLenum texture_target)
 {
     switch (texture_target) {
     case GL_TEXTURE_2D:
         m_texture_target_2d = nullptr;
-        m_currently_bound_target = 0;
+        m_currently_bound_target = GL_NONE;
+        m_texture_2d_enabled = false;
```

Comment:
Same here

Reply:
Actually the whole function should be removed eventually. You don't ever "unbind" a texture in OpenGL. Calling `glBindTexture(GL_TEXTURE_2D, 0)` binds the "default texture" to the texture unit. Sampling the default texture returns `RGBA(0,0,0,1)`.

https://stackoverflow.com/questions/15273674/binding-a-zero-texture-in-opengl

Reply:
I agree - I've added a commit that removes `TextureUnit::unbind_texture()` in favor of calling `::bind_texture_to_target()` with a `nullptr` texture.